docs/robot_locators.rst

```diff
@@ -0,0 +1,205 @@
+keyword. While it’s possible to write your own strategies using
+keywords, the locator manager makes it easy to associate a locator
+prefix with a dictionary of locators.
+
+This registration is done via the ``register_locators`` method of the
+locator manager, and should be done in the ``__init__`` method of a
+keyword library.
+
+For example, here is what it might look like for a library that
+contains keywords for the calendar tab.
+
+.. code-block:: python
+
+   from robot.libraries.BuiltIn import BuiltIn
+   from cumulusci.robotframework import locator_manager
+
+   locators = {...}  # see previous example
+
+   class CalendarLibrary:
+       ROBOT_LIBRARY_SCOPE = "GLOBAL"
+
+       def __init__(self):
+           locator_manager.register_locators("calendar", locators)
+
+When this library is imported into a test case file, the prefix
+“calendar” will be registered with SeleniumLibrary as a custom locator
+strategy.
+
+Using custom locators
+---------------------
+
+Once the dictionary has been defined and has been registered with a
+prefix, the locators work very similarly to any other locator. Because
+the dictionaries can be nested, you can separate the levels with a
+period (ie: dot notation).
+
+For example, with our example locators the options button locator can
+be used like in the following example:
+
+.. code-block::
+
+   Click element   calendar:sidebar.options button
+
+
+The following table shows how the locator is parsed:
+
++--------------------+---------------------------------------------------------------------------------------+
+| ``calendar:``      | locator prefix                                                                        |
++--------------------+---------------------------------------------------------------------------------------+
+| ``sidebar``        | first level of the dictionary (eg: ``locators['sidebar']``)                           |
++--------------------+---------------------------------------------------------------------------------------+
+| ``.``              | a level separator                                                                     |
++--------------------+---------------------------------------------------------------------------------------+
+| ``options button`` | the next level of a nested dictionary (eg: ``locators['sidebar']['options_button']``) |
++--------------------+---------------------------------------------------------------------------------------+
+
+
+Parameterized Locators
+----------------------
+
+Sometimes the only difference between multiple elements on a page is
+the text displayed in that element. For example, the html markup for a
+save, edit, and cancel button may be identical except for the word
+"Save", "Edit", or "Cancel".
+
+While you could create three separate locators for these three
+buttons, using a parameterized locator means we can replace three
+locators with one, which helps to keep our tech debt under control.
```

Comment:
Suddenly, I'm not sure who the audience is. Is this an internal-only doc or an external developer doc? If it's external, instead of mentioning tech debt, we might want to say something like:

``` 
While you can create a separate locator for each button, it's better to use a single parameterized locator for multiple buttons, which gives you more flexibility.
```
(or whatever the benefit is).

Reply:
The audience is largely going to be any person writing tests on top of Salesforce. I like your suggestion, I'll make the change. Thanks!